sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/image/image_search_space_utils.py

```diff
@@ -158,13 +165,43 @@ def _convert_sweep_dist_dict_to_str_dict(sweep_distribution: dict) -> dict:
     return sweep_distribution
 
 
+class ChoicePlusSchema(ChoiceSchema):
+    """ Choice schema that allows boolean values also"""
+    values = fields.List(
+        UnionField(
+            [
+                DumpableIntegerField(strict=True),
+                DumpableStringField(),
+                fields.Float(),
+                fields.Boolean(),
+                fields.Dict(
+                    keys=fields.Str(),
+                    values=UnionField(
+                        [
+                            NestedField("ChoiceSchema"),
+                            NestedField(NormalSchema()),
+                            NestedField(QNormalSchema()),
+                            NestedField(RandintSchema()),
+                            NestedField(UniformSchema()),
+                            NestedField(QUniformSchema()),
+                            DumpableIntegerField(strict=True),
+                            fields.Float(),
+                            fields.Str(),
+                            fields.Boolean(),
+                        ]
+                    ),
+                ),
+            ]
+        )
+    )
+
 def _convert_sweep_dist_dict_item_to_str(sweep_distribution: Union[bool, int, float, str, dict]) -> str:
     # Convert a Sweep Distribution dict to Sweep Distribution string
     # Eg. {type: 'choice', values: ['vitb16r224','vits16r224']} => "Choice('vitb16r224','vits16r224')"
     if isinstance(sweep_distribution, dict):
         sweep_dist_type = sweep_distribution["type"]
         if sweep_dist_type == SearchSpace.CHOICE:
-            sweep_dist_obj = ChoiceSchema().load(sweep_distribution) # pylint: disable=no-member
```

Comment:
Since we are modifying how choice values are parsed in pipelines workflow, please test choice with all types (bool, int, string, float) work well with you changes. Please share run ids as well.

Reply:
Sure Phani. Below are the sample runs.

**Choice of Ints**
number_of_epochs: choice([1,2,3,4,5])
https://ml.azure.com/experiments/id/1301f4ab-0d35-4d0f-ab02-b6efb926c6b5/runs/great_cheese_ngkmtc09bs?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/nilesh-new-rg/providers/Microsoft.MachineLearningServices/workspaces/nilesh-ws-new&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

**Choice of floats and strings**
learning_rate: choice(0.01, 0.001)
model_name: choice(vitb16r224, vits16r224)
https://ml.azure.com/experiments/id/1301f4ab-0d35-4d0f-ab02-b6efb926c6b5/runs/amiable_foot_vqkz8fhfv6?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/nilesh

**choice of boolean**
early_stopping: choice(False, True)
https://ml.azure.com/experiments/id/1301f4ab-0d35-4d0f-ab02-b6efb926c6b5/runs/mighty_beet_06xss9pxkm?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/nilesh-new-rg/providers/Microsoft.MachineLearningServices/workspaces/nilesh-ws-new&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#